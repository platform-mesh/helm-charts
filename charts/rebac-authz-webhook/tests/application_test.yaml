# yaml-language-server: $schema=https://raw.githubusercontent.com/helm-unittest/helm-unittest/main/schema/helm-testsuite.json
suite: Test the application manifests
values:
  - ../test-values.yaml
chart:
  version: 0.0.0
  appVersion: "0.0.0"
release:
  name: rebac-authz-webhook
  namespace: openmfp-system
tests:
- it: it should render the application manifests
  asserts:
  - matchSnapshot: {}
- it: it should render the application manifests with service monitor
  set:
    serviceMonitor:
      enabled: true
  asserts:
  - matchSnapshot: {}
- it: it should render the application manifests with service account
  set:
    serviceAccount:
      create: true
      name: "test"
  asserts:
  - matchSnapshot: {}
- it: it should render the application manifests with useInClusterClient
  set:
    useInClusterClient: true
  asserts:
  - matchSnapshot: {}
- it: it should render the cert-manager manifests with certManager enabled
  set:
    useInClusterClient: true
    global:
      istio:
        enabled: true
      certManager:
        enabled: true
    service:
      clusterIP: "10.96.86.219"
    istio:
      exposed: true
      dnsNames:
      - webhook.portal.cc.showroom.apeirora.eu
  asserts:
  - matchSnapshot: {}
- it: has all health checks
  template: deployment.yaml
  asserts:
    - equal:
        path: spec.template.spec.containers[0].livenessProbe.httpGet.path
        value: /healthz
    - equal:
        path: spec.template.spec.containers[0].readinessProbe.httpGet.path
        value: /readyz
    - equal:
        path: spec.template.spec.containers[0].startupProbe.httpGet.path
        value: /readyz

