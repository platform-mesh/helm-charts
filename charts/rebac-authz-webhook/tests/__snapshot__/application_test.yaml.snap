it should render the application manifests:
  1: |
    apiVersion: rbac.authorization.k8s.io/v1
    kind: ClusterRole
    metadata:
      name: rebac-authz-webhook
    rules:
      - apiGroups:
          - core.platform-mesh.io
        resources:
          - accounts
        verbs:
          - get
          - list
          - watch
      - apiGroups:
          - core.platform-mesh.io
        resources:
          - accounts
          - accounts/status
        verbs:
          - get
          - list
          - watch
      - apiGroups:
          - ""
        resources:
          - namespaces
        verbs:
          - get
          - list
          - watch
  2: |
    apiVersion: rbac.authorization.k8s.io/v1
    kind: ClusterRoleBinding
    metadata:
      name: rebac-authz-webhook
    roleRef:
      apiGroup: rbac.authorization.k8s.io
      kind: ClusterRole
      name: rebac-authz-webhook
    subjects:
      - kind: ServiceAccount
        name: rebac-authz-webhook
        namespace: openmfp-system
  3: |
    apiVersion: apps/v1
    kind: Deployment
    metadata:
      labels:
        app.kubernetes.io/instance: rebac-authz-webhook
        app.kubernetes.io/managed-by: Helm
        app.kubernetes.io/name: rebac-authz-webhook
        app.kubernetes.io/version: 0.0.0
        helm.sh/chart: rebac-authz-webhook-0.0.0
      name: rebac-authz-webhook
      namespace: openmfp-system
    spec:
      selector:
        matchLabels:
          app.kubernetes.io/instance: rebac-authz-webhook
          app.kubernetes.io/name: rebac-authz-webhook
      template:
        metadata:
          annotations: {}
          labels:
            app.kubernetes.io/instance: rebac-authz-webhook
            app.kubernetes.io/managed-by: Helm
            app.kubernetes.io/name: rebac-authz-webhook
            app.kubernetes.io/version: 0.0.0
            helm.sh/chart: rebac-authz-webhook-0.0.0
            networking.resources.gardener.cloud/to-openfga-tcp-grpc: allowed
        spec:
          containers:
            - args:
                - serve
                - --openfga-addr=https://example.com
                - --kcp-kubeconfig-path=/etc/kcp-kubeconfig/kubeconfig
                - --webhook-audience=rebac-authz-webhook
                - --log-level=INFO
                - --health-probe-bind-address=:8081
              env: null
              image: ghcr.io/platform-mesh/rebac-authz-webhook:0.0.0
              livenessProbe:
                failureThreshold: 1
                httpGet:
                  path: /healthz
                  port: 8081
                periodSeconds: 10
              name: rebac-authz-webhook
              ports:
                - containerPort: 9443
                - containerPort: 8080
              readinessProbe:
                httpGet:
                  path: /readyz
                  port: 8081
                initialDelaySeconds: 5
                periodSeconds: 10
              resources:
                limits:
                  cpu: 500m
                  memory: 128Mi
              startupProbe:
                failureThreshold: 30
                httpGet:
                  path: /readyz
                  port: 8081
                periodSeconds: 10
              volumeMounts:
                - mountPath: /config
                  name: certs
                - mountPath: /etc/kcp-kubeconfig
                  name: kcp-kubeconfig
                  readOnly: true
          imagePullSecrets:
            - name: github
          serviceAccountName: rebac-authz-webhook
          volumes:
            - name: certs
              secret:
                secretName: rebac-authz-webhook-cert
            - name: kcp-kubeconfig
              secret:
                secretName: rebac-authz-webhook-kubeconfig
  4: |
    apiVersion: cert-manager.io/v1
    kind: Issuer
    metadata:
      name: rebac-authz-webhook-issuer
      namespace: openmfp-system
    spec:
      ca:
        secretName: rebac-authz-webhook-webhook-ca
  5: |
    apiVersion: cert-manager.io/v1
    kind: Certificate
    metadata:
      name: rebac-authz-webhook-cert
      namespace: openmfp-system
    spec:
      dnsNames: []
      ipAddresses:
        - 10.96.86.219
      issuerRef:
        name: rebac-authz-webhook-issuer
      secretName: rebac-authz-webhook-cert
  6: |
    apiVersion: v1
    kind: Service
    metadata:
      labels:
        app.kubernetes.io/instance: rebac-authz-webhook
        app.kubernetes.io/managed-by: Helm
        app.kubernetes.io/name: rebac-authz-webhook
        app.kubernetes.io/version: 0.0.0
        helm.sh/chart: rebac-authz-webhook-0.0.0
      name: rebac-authz-webhook
    spec:
      ports:
        - name: rebac-authz-webhook
          port: 9443
          targetPort: 9443
        - name: metrics
          port: 8080
          targetPort: 8080
      selector:
        app.kubernetes.io/instance: rebac-authz-webhook
        app.kubernetes.io/name: rebac-authz-webhook
      type: ClusterIP
  7: |
    apiVersion: v1
    automountServiceAccountToken: true
    imagePullSecrets:
      - name: github
    kind: ServiceAccount
    metadata:
      labels:
        app.kubernetes.io/instance: rebac-authz-webhook
        app.kubernetes.io/managed-by: Helm
        app.kubernetes.io/name: rebac-authz-webhook
        app.kubernetes.io/version: 0.0.0
        helm.sh/chart: rebac-authz-webhook-0.0.0
      name: rebac-authz-webhook
it should render the application manifests with service account:
  1: |
    apiVersion: rbac.authorization.k8s.io/v1
    kind: ClusterRole
    metadata:
      name: rebac-authz-webhook
    rules:
      - apiGroups:
          - core.platform-mesh.io
        resources:
          - accounts
        verbs:
          - get
          - list
          - watch
      - apiGroups:
          - core.platform-mesh.io
        resources:
          - accounts
          - accounts/status
        verbs:
          - get
          - list
          - watch
      - apiGroups:
          - ""
        resources:
          - namespaces
        verbs:
          - get
          - list
          - watch
  2: |
    apiVersion: rbac.authorization.k8s.io/v1
    kind: ClusterRoleBinding
    metadata:
      name: rebac-authz-webhook
    roleRef:
      apiGroup: rbac.authorization.k8s.io
      kind: ClusterRole
      name: rebac-authz-webhook
    subjects:
      - kind: ServiceAccount
        name: rebac-authz-webhook
        namespace: openmfp-system
  3: |
    apiVersion: apps/v1
    kind: Deployment
    metadata:
      labels:
        app.kubernetes.io/instance: rebac-authz-webhook
        app.kubernetes.io/managed-by: Helm
        app.kubernetes.io/name: rebac-authz-webhook
        app.kubernetes.io/version: 0.0.0
        helm.sh/chart: rebac-authz-webhook-0.0.0
      name: rebac-authz-webhook
      namespace: openmfp-system
    spec:
      selector:
        matchLabels:
          app.kubernetes.io/instance: rebac-authz-webhook
          app.kubernetes.io/name: rebac-authz-webhook
      template:
        metadata:
          annotations: {}
          labels:
            app.kubernetes.io/instance: rebac-authz-webhook
            app.kubernetes.io/managed-by: Helm
            app.kubernetes.io/name: rebac-authz-webhook
            app.kubernetes.io/version: 0.0.0
            helm.sh/chart: rebac-authz-webhook-0.0.0
            networking.resources.gardener.cloud/to-openfga-tcp-grpc: allowed
        spec:
          containers:
            - args:
                - serve
                - --openfga-addr=https://example.com
                - --kcp-kubeconfig-path=/etc/kcp-kubeconfig/kubeconfig
                - --webhook-audience=rebac-authz-webhook
                - --log-level=INFO
                - --health-probe-bind-address=:8081
              env: null
              image: ghcr.io/platform-mesh/rebac-authz-webhook:0.0.0
              livenessProbe:
                failureThreshold: 1
                httpGet:
                  path: /healthz
                  port: 8081
                periodSeconds: 10
              name: rebac-authz-webhook
              ports:
                - containerPort: 9443
                - containerPort: 8080
              readinessProbe:
                httpGet:
                  path: /readyz
                  port: 8081
                initialDelaySeconds: 5
                periodSeconds: 10
              resources:
                limits:
                  cpu: 500m
                  memory: 128Mi
              startupProbe:
                failureThreshold: 30
                httpGet:
                  path: /readyz
                  port: 8081
                periodSeconds: 10
              volumeMounts:
                - mountPath: /config
                  name: certs
                - mountPath: /etc/kcp-kubeconfig
                  name: kcp-kubeconfig
                  readOnly: true
          imagePullSecrets:
            - name: github
          serviceAccountName: rebac-authz-webhook
          volumes:
            - name: certs
              secret:
                secretName: rebac-authz-webhook-cert
            - name: kcp-kubeconfig
              secret:
                secretName: rebac-authz-webhook-kubeconfig
  4: |
    apiVersion: cert-manager.io/v1
    kind: Issuer
    metadata:
      name: rebac-authz-webhook-issuer
      namespace: openmfp-system
    spec:
      ca:
        secretName: rebac-authz-webhook-webhook-ca
  5: |
    apiVersion: cert-manager.io/v1
    kind: Certificate
    metadata:
      name: rebac-authz-webhook-cert
      namespace: openmfp-system
    spec:
      dnsNames: []
      ipAddresses:
        - 10.96.86.219
      issuerRef:
        name: rebac-authz-webhook-issuer
      secretName: rebac-authz-webhook-cert
  6: |
    apiVersion: v1
    kind: Service
    metadata:
      labels:
        app.kubernetes.io/instance: rebac-authz-webhook
        app.kubernetes.io/managed-by: Helm
        app.kubernetes.io/name: rebac-authz-webhook
        app.kubernetes.io/version: 0.0.0
        helm.sh/chart: rebac-authz-webhook-0.0.0
      name: rebac-authz-webhook
    spec:
      ports:
        - name: rebac-authz-webhook
          port: 9443
          targetPort: 9443
        - name: metrics
          port: 8080
          targetPort: 8080
      selector:
        app.kubernetes.io/instance: rebac-authz-webhook
        app.kubernetes.io/name: rebac-authz-webhook
      type: ClusterIP
  7: |
    apiVersion: v1
    automountServiceAccountToken: true
    imagePullSecrets:
      - name: github
    kind: ServiceAccount
    metadata:
      labels:
        app.kubernetes.io/instance: rebac-authz-webhook
        app.kubernetes.io/managed-by: Helm
        app.kubernetes.io/name: rebac-authz-webhook
        app.kubernetes.io/version: 0.0.0
        helm.sh/chart: rebac-authz-webhook-0.0.0
      name: rebac-authz-webhook
it should render the application manifests with service monitor:
  1: |
    apiVersion: rbac.authorization.k8s.io/v1
    kind: ClusterRole
    metadata:
      name: rebac-authz-webhook
    rules:
      - apiGroups:
          - core.platform-mesh.io
        resources:
          - accounts
        verbs:
          - get
          - list
          - watch
      - apiGroups:
          - core.platform-mesh.io
        resources:
          - accounts
          - accounts/status
        verbs:
          - get
          - list
          - watch
      - apiGroups:
          - ""
        resources:
          - namespaces
        verbs:
          - get
          - list
          - watch
  2: |
    apiVersion: rbac.authorization.k8s.io/v1
    kind: ClusterRoleBinding
    metadata:
      name: rebac-authz-webhook
    roleRef:
      apiGroup: rbac.authorization.k8s.io
      kind: ClusterRole
      name: rebac-authz-webhook
    subjects:
      - kind: ServiceAccount
        name: rebac-authz-webhook
        namespace: openmfp-system
  3: |
    apiVersion: apps/v1
    kind: Deployment
    metadata:
      labels:
        app.kubernetes.io/instance: rebac-authz-webhook
        app.kubernetes.io/managed-by: Helm
        app.kubernetes.io/name: rebac-authz-webhook
        app.kubernetes.io/version: 0.0.0
        helm.sh/chart: rebac-authz-webhook-0.0.0
      name: rebac-authz-webhook
      namespace: openmfp-system
    spec:
      selector:
        matchLabels:
          app.kubernetes.io/instance: rebac-authz-webhook
          app.kubernetes.io/name: rebac-authz-webhook
      template:
        metadata:
          annotations: {}
          labels:
            app.kubernetes.io/instance: rebac-authz-webhook
            app.kubernetes.io/managed-by: Helm
            app.kubernetes.io/name: rebac-authz-webhook
            app.kubernetes.io/version: 0.0.0
            helm.sh/chart: rebac-authz-webhook-0.0.0
            networking.resources.gardener.cloud/to-openfga-tcp-grpc: allowed
        spec:
          containers:
            - args:
                - serve
                - --openfga-addr=https://example.com
                - --kcp-kubeconfig-path=/etc/kcp-kubeconfig/kubeconfig
                - --webhook-audience=rebac-authz-webhook
                - --log-level=INFO
                - --health-probe-bind-address=:8081
              env: null
              image: ghcr.io/platform-mesh/rebac-authz-webhook:0.0.0
              livenessProbe:
                failureThreshold: 1
                httpGet:
                  path: /healthz
                  port: 8081
                periodSeconds: 10
              name: rebac-authz-webhook
              ports:
                - containerPort: 9443
                - containerPort: 8080
              readinessProbe:
                httpGet:
                  path: /readyz
                  port: 8081
                initialDelaySeconds: 5
                periodSeconds: 10
              resources:
                limits:
                  cpu: 500m
                  memory: 128Mi
              startupProbe:
                failureThreshold: 30
                httpGet:
                  path: /readyz
                  port: 8081
                periodSeconds: 10
              volumeMounts:
                - mountPath: /config
                  name: certs
                - mountPath: /etc/kcp-kubeconfig
                  name: kcp-kubeconfig
                  readOnly: true
          imagePullSecrets:
            - name: github
          serviceAccountName: rebac-authz-webhook
          volumes:
            - name: certs
              secret:
                secretName: rebac-authz-webhook-cert
            - name: kcp-kubeconfig
              secret:
                secretName: rebac-authz-webhook-kubeconfig
  4: |
    apiVersion: cert-manager.io/v1
    kind: Issuer
    metadata:
      name: rebac-authz-webhook-issuer
      namespace: openmfp-system
    spec:
      ca:
        secretName: rebac-authz-webhook-webhook-ca
  5: |
    apiVersion: cert-manager.io/v1
    kind: Certificate
    metadata:
      name: rebac-authz-webhook-cert
      namespace: openmfp-system
    spec:
      dnsNames: []
      ipAddresses:
        - 10.96.86.219
      issuerRef:
        name: rebac-authz-webhook-issuer
      secretName: rebac-authz-webhook-cert
  6: |
    apiVersion: v1
    kind: Service
    metadata:
      labels:
        app.kubernetes.io/instance: rebac-authz-webhook
        app.kubernetes.io/managed-by: Helm
        app.kubernetes.io/name: rebac-authz-webhook
        app.kubernetes.io/version: 0.0.0
        helm.sh/chart: rebac-authz-webhook-0.0.0
      name: rebac-authz-webhook
    spec:
      ports:
        - name: rebac-authz-webhook
          port: 9443
          targetPort: 9443
        - name: metrics
          port: 8080
          targetPort: 8080
      selector:
        app.kubernetes.io/instance: rebac-authz-webhook
        app.kubernetes.io/name: rebac-authz-webhook
      type: ClusterIP
  7: |
    apiVersion: v1
    automountServiceAccountToken: true
    imagePullSecrets:
      - name: github
    kind: ServiceAccount
    metadata:
      labels:
        app.kubernetes.io/instance: rebac-authz-webhook
        app.kubernetes.io/managed-by: Helm
        app.kubernetes.io/name: rebac-authz-webhook
        app.kubernetes.io/version: 0.0.0
        helm.sh/chart: rebac-authz-webhook-0.0.0
      name: rebac-authz-webhook
  8: |
    apiVersion: monitoring.coreos.com/v1
    kind: ServiceMonitor
    metadata:
      labels:
        app.kubernetes.io/instance: rebac-authz-webhook
        app.kubernetes.io/managed-by: Helm
        app.kubernetes.io/name: rebac-authz-webhook
        app.kubernetes.io/version: 0.0.0
        helm.sh/chart: rebac-authz-webhook-0.0.0
        prometheus: seed
      name: rebac-authz-webhook
    spec:
      endpoints:
        - port: metrics
      selector:
        matchLabels:
          app.kubernetes.io/instance: rebac-authz-webhook
          app.kubernetes.io/managed-by: Helm
          app.kubernetes.io/name: rebac-authz-webhook
          app.kubernetes.io/version: 0.0.0
          helm.sh/chart: rebac-authz-webhook-0.0.0
it should render the application manifests with useInClusterClient:
  1: |
    apiVersion: rbac.authorization.k8s.io/v1
    kind: ClusterRole
    metadata:
      name: rebac-authz-webhook
    rules:
      - apiGroups:
          - core.platform-mesh.io
        resources:
          - accounts
        verbs:
          - get
          - list
          - watch
      - apiGroups:
          - core.platform-mesh.io
        resources:
          - accounts
          - accounts/status
        verbs:
          - get
          - list
          - watch
      - apiGroups:
          - ""
        resources:
          - namespaces
        verbs:
          - get
          - list
          - watch
  2: |
    apiVersion: rbac.authorization.k8s.io/v1
    kind: ClusterRoleBinding
    metadata:
      name: rebac-authz-webhook
    roleRef:
      apiGroup: rbac.authorization.k8s.io
      kind: ClusterRole
      name: rebac-authz-webhook
    subjects:
      - kind: ServiceAccount
        name: rebac-authz-webhook
        namespace: openmfp-system
  3: |
    apiVersion: apps/v1
    kind: Deployment
    metadata:
      labels:
        app.kubernetes.io/instance: rebac-authz-webhook
        app.kubernetes.io/managed-by: Helm
        app.kubernetes.io/name: rebac-authz-webhook
        app.kubernetes.io/version: 0.0.0
        helm.sh/chart: rebac-authz-webhook-0.0.0
      name: rebac-authz-webhook
      namespace: openmfp-system
    spec:
      selector:
        matchLabels:
          app.kubernetes.io/instance: rebac-authz-webhook
          app.kubernetes.io/name: rebac-authz-webhook
      template:
        metadata:
          annotations: {}
          labels:
            app.kubernetes.io/instance: rebac-authz-webhook
            app.kubernetes.io/managed-by: Helm
            app.kubernetes.io/name: rebac-authz-webhook
            app.kubernetes.io/version: 0.0.0
            helm.sh/chart: rebac-authz-webhook-0.0.0
            networking.resources.gardener.cloud/to-openfga-tcp-grpc: allowed
        spec:
          containers:
            - args:
                - serve
                - --openfga-addr=https://example.com
                - --kcp-kubeconfig-path=/etc/kcp-kubeconfig/kubeconfig
                - --webhook-audience=rebac-authz-webhook
                - --log-level=INFO
                - --health-probe-bind-address=:8081
              env: null
              image: ghcr.io/platform-mesh/rebac-authz-webhook:0.0.0
              livenessProbe:
                failureThreshold: 1
                httpGet:
                  path: /healthz
                  port: 8081
                periodSeconds: 10
              name: rebac-authz-webhook
              ports:
                - containerPort: 9443
                - containerPort: 8080
              readinessProbe:
                httpGet:
                  path: /readyz
                  port: 8081
                initialDelaySeconds: 5
                periodSeconds: 10
              resources:
                limits:
                  cpu: 500m
                  memory: 128Mi
              startupProbe:
                failureThreshold: 30
                httpGet:
                  path: /readyz
                  port: 8081
                periodSeconds: 10
              volumeMounts:
                - mountPath: /config
                  name: certs
                - mountPath: /etc/kcp-kubeconfig
                  name: kcp-kubeconfig
                  readOnly: true
          imagePullSecrets:
            - name: github
          serviceAccountName: rebac-authz-webhook
          volumes:
            - name: certs
              secret:
                secretName: rebac-authz-webhook-cert
            - name: kcp-kubeconfig
              secret:
                secretName: rebac-authz-webhook-kubeconfig
  4: |
    apiVersion: cert-manager.io/v1
    kind: Issuer
    metadata:
      name: rebac-authz-webhook-issuer
      namespace: openmfp-system
    spec:
      ca:
        secretName: rebac-authz-webhook-webhook-ca
  5: |
    apiVersion: cert-manager.io/v1
    kind: Certificate
    metadata:
      name: rebac-authz-webhook-cert
      namespace: openmfp-system
    spec:
      dnsNames: []
      ipAddresses:
        - 10.96.86.219
      issuerRef:
        name: rebac-authz-webhook-issuer
      secretName: rebac-authz-webhook-cert
  6: |
    apiVersion: v1
    kind: Service
    metadata:
      labels:
        app.kubernetes.io/instance: rebac-authz-webhook
        app.kubernetes.io/managed-by: Helm
        app.kubernetes.io/name: rebac-authz-webhook
        app.kubernetes.io/version: 0.0.0
        helm.sh/chart: rebac-authz-webhook-0.0.0
      name: rebac-authz-webhook
    spec:
      ports:
        - name: rebac-authz-webhook
          port: 9443
          targetPort: 9443
        - name: metrics
          port: 8080
          targetPort: 8080
      selector:
        app.kubernetes.io/instance: rebac-authz-webhook
        app.kubernetes.io/name: rebac-authz-webhook
      type: ClusterIP
  7: |
    apiVersion: v1
    automountServiceAccountToken: true
    imagePullSecrets:
      - name: github
    kind: ServiceAccount
    metadata:
      labels:
        app.kubernetes.io/instance: rebac-authz-webhook
        app.kubernetes.io/managed-by: Helm
        app.kubernetes.io/name: rebac-authz-webhook
        app.kubernetes.io/version: 0.0.0
        helm.sh/chart: rebac-authz-webhook-0.0.0
      name: rebac-authz-webhook
it should render the cert-manager manifests with certManager enabled:
  1: |
    apiVersion: rbac.authorization.k8s.io/v1
    kind: ClusterRole
    metadata:
      name: rebac-authz-webhook
    rules:
      - apiGroups:
          - core.platform-mesh.io
        resources:
          - accounts
        verbs:
          - get
          - list
          - watch
      - apiGroups:
          - core.platform-mesh.io
        resources:
          - accounts
          - accounts/status
        verbs:
          - get
          - list
          - watch
      - apiGroups:
          - ""
        resources:
          - namespaces
        verbs:
          - get
          - list
          - watch
  2: |
    apiVersion: rbac.authorization.k8s.io/v1
    kind: ClusterRoleBinding
    metadata:
      name: rebac-authz-webhook
    roleRef:
      apiGroup: rbac.authorization.k8s.io
      kind: ClusterRole
      name: rebac-authz-webhook
    subjects:
      - kind: ServiceAccount
        name: rebac-authz-webhook
        namespace: openmfp-system
  3: |
    apiVersion: apps/v1
    kind: Deployment
    metadata:
      labels:
        app.kubernetes.io/instance: rebac-authz-webhook
        app.kubernetes.io/managed-by: Helm
        app.kubernetes.io/name: rebac-authz-webhook
        app.kubernetes.io/version: 0.0.0
        helm.sh/chart: rebac-authz-webhook-0.0.0
      name: rebac-authz-webhook
      namespace: openmfp-system
    spec:
      selector:
        matchLabels:
          app.kubernetes.io/instance: rebac-authz-webhook
          app.kubernetes.io/name: rebac-authz-webhook
      template:
        metadata:
          annotations: {}
          labels:
            app.kubernetes.io/instance: rebac-authz-webhook
            app.kubernetes.io/managed-by: Helm
            app.kubernetes.io/name: rebac-authz-webhook
            app.kubernetes.io/version: 0.0.0
            helm.sh/chart: rebac-authz-webhook-0.0.0
            networking.resources.gardener.cloud/to-openfga-tcp-grpc: allowed
        spec:
          containers:
            - args:
                - serve
                - --openfga-addr=https://example.com
                - --kcp-kubeconfig-path=/etc/kcp-kubeconfig/kubeconfig
                - --webhook-audience=rebac-authz-webhook
                - --log-level=INFO
                - --health-probe-bind-address=:8081
              env: null
              image: ghcr.io/platform-mesh/rebac-authz-webhook:0.0.0
              livenessProbe:
                failureThreshold: 1
                httpGet:
                  path: /healthz
                  port: 8081
                periodSeconds: 10
              name: rebac-authz-webhook
              ports:
                - containerPort: 9443
                - containerPort: 8080
              readinessProbe:
                httpGet:
                  path: /readyz
                  port: 8081
                initialDelaySeconds: 5
                periodSeconds: 10
              resources:
                limits:
                  cpu: 500m
                  memory: 128Mi
              startupProbe:
                failureThreshold: 30
                httpGet:
                  path: /readyz
                  port: 8081
                periodSeconds: 10
              volumeMounts:
                - mountPath: /config
                  name: certs
                - mountPath: /etc/kcp-kubeconfig
                  name: kcp-kubeconfig
                  readOnly: true
          imagePullSecrets:
            - name: github
          serviceAccountName: rebac-authz-webhook
          volumes:
            - name: certs
              secret:
                secretName: rebac-authz-webhook-cert
            - name: kcp-kubeconfig
              secret:
                secretName: rebac-authz-webhook-kubeconfig
  4: |
    apiVersion: cert-manager.io/v1
    kind: Issuer
    metadata:
      name: rebac-authz-webhook-issuer
      namespace: openmfp-system
    spec:
      ca:
        secretName: rebac-authz-webhook-webhook-ca
  5: |
    apiVersion: cert-manager.io/v1
    kind: Certificate
    metadata:
      name: rebac-authz-webhook-cert
      namespace: openmfp-system
    spec:
      dnsNames:
        - webhook.portal.cc.showroom.apeirora.eu
      ipAddresses:
        - 10.96.86.219
      issuerRef:
        name: rebac-authz-webhook-issuer
      secretName: rebac-authz-webhook-cert
  6: |
    apiVersion: v1
    kind: Service
    metadata:
      labels:
        app.kubernetes.io/instance: rebac-authz-webhook
        app.kubernetes.io/managed-by: Helm
        app.kubernetes.io/name: rebac-authz-webhook
        app.kubernetes.io/version: 0.0.0
        helm.sh/chart: rebac-authz-webhook-0.0.0
      name: rebac-authz-webhook
    spec:
      clusterIP: 10.96.86.219
      ports:
        - name: rebac-authz-webhook
          port: 9443
          targetPort: 9443
        - name: metrics
          port: 8080
          targetPort: 8080
      selector:
        app.kubernetes.io/instance: rebac-authz-webhook
        app.kubernetes.io/name: rebac-authz-webhook
      type: ClusterIP
  7: |
    apiVersion: v1
    automountServiceAccountToken: true
    imagePullSecrets:
      - name: github
    kind: ServiceAccount
    metadata:
      labels:
        app.kubernetes.io/instance: rebac-authz-webhook
        app.kubernetes.io/managed-by: Helm
        app.kubernetes.io/name: rebac-authz-webhook
        app.kubernetes.io/version: 0.0.0
        helm.sh/chart: rebac-authz-webhook-0.0.0
      name: rebac-authz-webhook
  8: |
    apiVersion: networking.istio.io/v1beta1
    kind: VirtualService
    metadata:
      name: rebac-authz-webhook
      namespace: openmfp-system
    spec:
      gateways:
        - openmfp-system/gateway
      hosts:
        - webhook.portal.cc.showroom.apeirora.eu
      tls:
        - match:
            - sniHosts:
                - webhook.portal.cc.showroom.apeirora.eu
          route:
            - destination:
                host: rebac-authz-webhook.openmfp-system.svc.cluster.local
                port:
                  number: 9443
